lib/ui/pages/score/score_page.dart

```diff
@@ -125,188 +142,160 @@ class _ScoreViewerPageState extends State<ScoreViewerPage> with TickerProviderSt
     super.dispose();
   }
 
-  _onPopupMenuSelect(int value) async {
-    switch (value) {
-      case 0:
-        _addScoreRankTask();
-        break;
-      case 1:
-        _addSearchCourseTypeTask();
-        break;
-      default:
-        break;
-    }
-  }
-
   @override
-  Widget build(BuildContext context) {
-    return DefaultTabController(
-      length: tabLabelList.length,
-      child: Scaffold(
-        appBar: AppBar(
-          title: Text(R.current.searchScore),
-          actions: [
-            if (courseScoreList.isNotEmpty)
-              PopupMenuButton<int>(
-                onSelected: (result) {
-                  setState(() {
-                    _onPopupMenuSelect(result);
-                  });
-                },
-                itemBuilder: (BuildContext context) => [
-                  PopupMenuItem(
-                    value: 0,
-                    child: Text(R.current.refresh),
-                  ),
-                  PopupMenuItem(
-                    value: 1,
-                    child: Text(R.current.calculationCredit),
-                  ),
-                ],
-              ),
-          ],
-          bottom: TabBar(
-            controller: _tabController,
-            labelColor: AppColors.mainColor,
-            unselectedLabelColor: Colors.white,
-            indicatorSize: TabBarIndicatorSize.label,
-//      labelPadding: EdgeInsets.symmetric(horizontal: 8),
-            indicator: const BoxDecoration(
-              borderRadius: BorderRadius.only(
-                topLeft: Radius.circular(10),
-                topRight: Radius.circular(10),
+  Widget build(BuildContext context) => DefaultTabController(
+        length: tabLabelList.length,
+        child: Scaffold(
+          appBar: AppBar(
+            title: Text(R.current.searchScore),
+            actions: [
+              if (courseScoreList.isNotEmpty)
+                IconButton(
+                  onPressed: _addScoreRankTask,
+                  icon: const Icon(Icons.refresh),
+                ),
+              if (courseScoreList.isNotEmpty)
+                IconButton(
+                  onPressed: _addSearchCourseTypeTask,
+                  icon: const Icon(Icons.calculate_rounded),
+                ),
+            ],
+            bottom: TabBar(
+              controller: _tabController,
+              labelColor: AppColors.mainColor,
+              unselectedLabelColor: Colors.white,
+              indicatorSize: TabBarIndicatorSize.label,
+              indicator: const BoxDecoration(
+                borderRadius: BorderRadius.only(
+                  topLeft: Radius.circular(10),
+                  topRight: Radius.circular(10),
+                ),
+                color: Colors.white,
               ),
-              color: Colors.white,
+              isScrollable: true,
+              tabs: tabLabelList,
+              onTap: (int index) {
+                _currentTabIndex = index;
+                setState(() {});
```

Comment:
[Question]

Why not wrap the `_currentTabIndex = index;` statement inside `setState`?

Reply:
Actually I don't know, since it is created by the former maintainer. But this is actually a bad practice for using `setState`.

Reply:
Through this PR does not focus on refactor all parts of score page, we may fix these problems (which caused by the former maintainer) later.